src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -284,311 +338,19 @@ public void Save(ModelSaveContext ctx)
             }
         }
 
-        public const string LoaderSignature = "LdaTransform";
-        private static VersionInfo GetVersionInfo()
-        {
-            return new VersionInfo(
-                modelSignature: "LIGHTLDA",
-                verWrittenCur: 0x00010001, // Initial
-                verReadableCur: 0x00010001,
-                verWeCanReadBack: 0x00010001,
-                loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(LdaTransform).Assembly.FullName);
-        }
-
-        private readonly ColInfoEx[] _exes;
-        private readonly LdaState[] _ldas;
-        private readonly ColumnType[] _types;
-        private readonly bool _saveText;
-
-        private const string RegistrationName = "LightLda";
-        private const string WordTopicModelFilename = "word_topic_summary.txt";
-        internal const string Summary = "The LDA transform implements LightLDA, a state-of-the-art implementation of Latent Dirichlet Allocation.";
-        internal const string UserName = "Latent Dirichlet Allocation Transform";
-        internal const string ShortName = "LightLda";
-
-        public LdaTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(env, RegistrationName, args.Column, input, TestType)
+        public class LdaTopicSummary
         {
-            Host.CheckValue(args, nameof(args));
-            Host.CheckUserArg(args.NumTopic > 0, nameof(args.NumTopic), "Must be positive.");
-            Host.CheckValue(input, nameof(input));
-            Host.CheckUserArg(Utils.Size(args.Column) > 0, nameof(args.Column));
-            _exes = new ColInfoEx[Infos.Length];
-            _types = new ColumnType[Infos.Length];
-            _ldas = new LdaState[Infos.Length];
-            _saveText = args.OutputTopicWordSummary;
-            for (int i = 0; i < Infos.Length; i++)
-            {
-                var ex = new ColInfoEx(Host, args.Column[i], args);
-                _exes[i] = ex;
-                _types[i] = new VectorType(NumberType.Float, ex.NumTopic);
-            }
-            using (var ch = Host.Start("Train"))
-            {
-                Train(ch, input, _ldas);
-            }
-            Metadata.Seal();
-        }
+            public Dictionary<int, KeyValuePair<int, float>[]> SummaryVectorPerTopic;
```

Comment:
>public Dictionary<int, KeyValuePair<int, float>[]> [](start = 12, length = 50)

This is not a good design. The class should be immutable, and all the fields should be initialized in the ctor. The dictionary should also be readonly.

By the way, what does 'summary vector per topic' actually mean? I am completely in the dark of how this is supposed to be interpreted.

Reply:
I renamed the field to WordScoresPerTopic

for each topic discovered by LDA,  users may be interested in knowing the  words  within each topic  and their corresponding probabilities.  this is provided by WordScoresPerTopic

Currently for the (**word**, score) pairs  we are using a KeyValuePair<**int**, float>[] i.e. representing words as **int**   

Issue #1411  was created to look into how to  map the **int** to the proper word

---
In reply to: [233583415](https://github.com/dotnet/machinelearning/pull/1410#discussion_r233583415) [](ancestors = 233583415)